src/couch_httpd.erl

```diff
@@ -872,7 +872,7 @@ error_info({bad_ctype, Reason}) ->
     {415, <<"bad_content_type">>, Reason};
 error_info(requested_range_not_satisfiable) ->
     {416, <<"requested_range_not_satisfiable">>, <<"Requested range not satisfiable">>};
-error_info({error, illegal_database_name, Name}) ->
+error_info({error, {illegal_database_name, Name}}) ->
```

Comment:
I think the first option is the most sane and correct - validate function is only holds the knowledge about database name restrictions and throwing error with descriptive message  from there is correct. chttpd need just transform this error into HTTP response.